app/src/pages/models/CloneModelButton.tsx

```diff
@@ -0,0 +1,215 @@
+              cacheRead
+              cacheWrite
+              promptAudio
+              completionAudio
+              reasoning
+            }
+          }
+          __typename
+        }
+      }
+    `);
+  const notifySuccess = useNotifySuccess();
+  const notifyError = useNotifyError();
+  const modelData = data?.node;
+
+  if (!modelData) {
+    return <Alert variant="danger">Model not found</Alert>;
+  }
+
+  return (
+    <ModelForm
+      modelName={`${modelData.name} (override)`}
+      modelProvider={modelData.provider}
+      modelNamePattern={modelData.namePattern}
+      modelCost={modelData.tokenCost}
+      onSubmit={(params) => {
+        commitCloneModel({
+          variables: {
+            input: {
+              name: params.name,
+              provider: params.provider,
+              namePattern: params.namePattern,
+              inputCostPerToken: params.cost.input,
+              outputCostPerToken: params.cost.output,
+              cacheReadCostPerToken: params.cost.cacheRead,
+              cacheWriteCostPerToken: params.cost.cacheWrite,
+              promptAudioCostPerToken: params.cost.promptAudio,
+              completionAudioCostPerToken: params.cost.completionAudio,
+              reasoningCostPerToken: params.cost.reasoning,
+            },
+          },
+          onCompleted: () => {
+            onClose();
+            onModelCloned && onModelCloned(params);
+            notifySuccess({
+              title: `Model Cloned`,
+              message: `Model "${params.name}" cloned successfully`,
+            });
+          },
+          onError: (error) => {
+            const formattedError =
+              getErrorMessagesFromRelayMutationError(error);
+            notifyError({
+              title: "Failed to clone model",
+              message: formattedError?.[0] ?? "Failed to clone model",
+            });
+          },
+        });
+      }}
+      isSubmitting={isCommittingCloneModel}
+      submitButtonText="Save Changes"
+      formMode="create"
+    />
+  );
+}
+
+export function CloneModelButton({
+  modelId,
+  onModelCloned,
+}: {
+  modelId: string;
+  onModelCloned?: (model: ModelFormParams) => void;
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [queryReference, loadQuery, disposeQuery] =
+    useQueryLoader<CloneModelButtonQuery>(ModelQuery);
+
+  const handleOpen = () => {
+    loadQuery({ id: modelId }, { fetchPolicy: "network-only" });
+    setIsOpen(true);
+  };
+
+  const handleClose = () => {
+    setIsOpen(false);
+    disposeQuery();
+  };
+
+  return (
+    <DialogTrigger
+      isOpen={isOpen}
+      onOpenChange={(open) => {
+        if (!open) handleClose();
+      }}
+    >
+      <Button
+        variant="default"
+        leadingVisual={<Icon svg={<Icons.DuplicateIcon />} />}
+        aria-label="Clone model"
+        onPress={handleOpen}
+        size="S"
+      >
+        Clone
+      </Button>
+      <DialogContainer onDismiss={handleClose}>
```

Comment:
This component should not be used, checkout Dialog.stories.tsx for an example of dialog composition